cyral/resource_cyral_repository_identity_map.go

```diff
@@ -218,37 +232,63 @@ func resourceRepositoryIdentityMap(deprecate bool) *schema.Resource {
 						d.Get("identity_name").(string),
 						d.Get("repository_local_account_id").(string))
 				},
-				ResourceData: &RepositoryIdentityMapResource{},
-				ResponseData: &RepositoryIdentityMapAPIResponse{},
+				ResourceData:   &RepositoryIdentityMapResource{},
+				ResponseData:   &RepositoryIdentityMapAPIResponse{},
+				UnsetIdOnError: true,
 			}, ReadRepositoryIdentityMapConfig,
 		),
 		ReadContext: ReadResource(ReadRepositoryIdentityMapConfig),
-		UpdateContext: UpdateResource(
-			ResourceOperationConfig{
-				Name:       "RepositoryIdentityMapResourceUpdate",
-				HttpMethod: http.MethodPut,
-				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
-					return fmt.Sprintf("https://%s/v1/repos/%s/identityMaps/%s/%s/%s",
-						c.ControlPlane,
-						d.Get("repository_id").(string),
-						d.Get("identity_type").(string),
-						d.Get("identity_name").(string),
-						d.Get("repository_local_account_id").(string))
-				},
-				ResourceData: &RepositoryIdentityMapResource{},
-			}, ReadRepositoryIdentityMapConfig,
-		),
+		UpdateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			c := m.(*client.Client)
+			readConfig := ReadRepositoryIdentityMapConfig
+
+			repoIdNew := d.Get("repository_id")
+			idTypeNew := d.Get("identity_type")
+			idNameNew := d.Get("identity_name")
+			accIdNew := d.Get("repository_local_account_id")
+
+			resourceData := &RepositoryIdentityMapResource{}
+			resourceData.ReadFromSchema(d)
+
+			createNewUrl := fmt.Sprintf("https://%s/v1/repos/%s/identityMaps/%s/%s/%s",
+				c.ControlPlane,
+				repoIdNew,
+				idTypeNew,
+				idNameNew,
+				accIdNew,
+			)
+
+			if _, err := c.DoRequest(createNewUrl, http.MethodPut, resourceData); err != nil {
+				return createError("Unable to create new integration", fmt.Sprintf("%v", err))
+			}
+
+			d.SetId(fmt.Sprintf("%s/identityMaps/%s/%s/%s",
+				d.Get("repository_id").(string),
+				d.Get("identity_type").(string),
+				d.Get("identity_name").(string),
+				d.Get("repository_local_account_id").(string)),
+			)
+
+			removeUrl := fmt.Sprintf("https://%s/v1/repos/%s",
+				c.ControlPlane,
+				d.Id(),
+			)
+
+			if _, err := c.DoRequest(removeUrl, http.MethodDelete, resourceData); err != nil {
+				return createError("Unable to delete previous integration", fmt.Sprintf("%v", err))
+			}
+
+			return ReadResource(readConfig)(ctx, d, m)
+		},
+
 		DeleteContext: DeleteResource(
 			ResourceOperationConfig{
 				Name:       "RepositoryIdentityMapResourceDelete",
 				HttpMethod: http.MethodDelete,
 				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
-					return fmt.Sprintf("https://%s/v1/repos/%s/identityMaps/%s/%s/%s",
+					return fmt.Sprintf("https://%s/v1/repos/%s",
```

Comment:
I don't think it is a good idea to define the URL from the ID as these are different concepts. It also makes the code confusing and one may think that the URL is `https://CP_ADDRESS/v1/repos/SOME_ID` and just realize it the actual endpoint is different if the resource id is actually analyzed.

Reply:
If the url is not defined from the ID everything breaks because of what we've discussed over time. There's no way to not attach those two, since the url depends on the id, and the url is unique in the same way as the id.